src/algorithms/synthesis/dd_synthesis.cpp

```diff
@@ -584,6 +584,38 @@ namespace syrec {
 
         extend(tt);
 
+        // Refer to the one-pass synthesis  algorithm of https://www.cda.cit.tum.de/files/eda/2017_tcad_one_pass_synthesis_reversible_circuits.pdf.
+        if (onePass) {
+            if (m > n) {
+                // zeros are inserted to match the length of the output patterns.
+                augmentWithConstants(tt, m);
+            }
+
+            // k -> Minimum no. of additional lines required (no encoding is performed).
+            const auto [_, k] = encodeHuffman(tt, true);
+
+            const auto maxIO = std::max(n, m);
+
+            r           = (m + k) - maxIO;
+            totalNoBits = maxIO + r;
```

Comment:
Isn't that virtually the same computation as down below?
if yes, then move this out of the `if`.

Reply:
r is different for one pass. The others are the same.